src/brod_consumer.erl

```diff
@@ -48,265 +42,370 @@
         , handle_info/2
         , terminate/2
         , code_change/3
-        , format_status/2
         ]).
 
 -include("brod.hrl").
 -include("brod_int.hrl").
 
--record(state, { hosts         :: [{string(), integer()}]
-               , socket        :: #socket{}
-               , topic         :: binary()
-               , partition     :: integer()
-               , callback      :: callback_fun()
-               , offset        :: integer()
-               , max_wait_time :: integer()
-               , min_bytes     :: integer()
-               , max_bytes     :: integer()
-               , sleep_timeout :: integer()
+-type consumer_option() :: offset
+                         | min_bytes
+                         | max_bytes
+                         | max_wait_time
+                         | sleep_timeout
+                         | prefetch_count.
+
+%% behaviour definition
+-callback init_consumer( Topic     :: topic()
```

Comment:
add more docs?

btw, why don't return CbOptions (or NewCbOptions) as the second element in the returned tuple ?